fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -1896,30 +1896,31 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if Firefox Suggest is enabled.
      */
-    var enableFxSuggest by lazyFeatureFlagPreference(
+    var enableFxSuggest by featureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_fxsuggest),
-        default = { FxNimbus.features.fxSuggest.value().enabled },
+        default = FxNimbus.features.fxSuggest.value().enabled,
         featureFlag = FeatureFlags.fxSuggest,
     )
 
     /**
      * Indicates if the user has chosen to show sponsored search suggestions in the awesomebar.
-     * The default value is computed lazily, and based on whether Firefox Suggest is enabled.
+     * Defaults to `true` if Firefox Suggest is enabled and the user has not explicitly changed the
+     * setting. Always `false` and can't be changed if Firefox Suggest is disabled.
      */
-    var showSponsoredSuggestions by lazyFeatureFlagPreference(
+    var showSponsoredSuggestions by featureFlagPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_sponsored_suggestions),
-        default = { enableFxSuggest },
-        featureFlag = FeatureFlags.fxSuggest,
+        default = true,
+        featureFlag = enableFxSuggest,
```

Comment:
This is a "creative" use of `featureFlagPreference`, and I wanted to drop a comment with some more details why—and maybe see if there's a better way of doing that?

I think the behavior we want is:

* If `enableFxSuggest` is `true`, then `showSponsoredSuggestions` (and `showNonSponsoredSuggestions`) should be the user-set value. If there's no user-set value, `show{Non}SponsoredSuggestions` should default to `true`.
* If `enableFxSuggest` is `false`, then `show{Non}SponsoredSuggestions` should always be `false`—regardless of whether they've previously had a user-set value or not.

`enableFxSuggest` can be `true` in three ways:

* You're on a Nightly or Debug build, or...
* You're enrolled in the Nimbus experiment, or...
* You turn on the secret setting.

In other cases, it's `false`.

Reply:
Another approach would be to change all places that check `settings.show{Non}SponsoredSuggestions` to check `settings.enableFxSuggest && settings.show{Non}SponsoredSuggestions`—but this felt slightly nicer to me 😊

Reply:
I imagine this logic is pretty subject to change as rollouts continue, so I'm happy landing this as is. That said, a couple alternative suggestions if you are interested in them:

- since `enableFxSuggest` seems outside of user control entirely, we could remove it as a cached preference and read from nimbus / feature flag directly. this could be abstracted into a function instead, like with `isAddressFeatureEnabled()`
- all of the logic could be converted to a function, something like `amoCollectionOverrideConfigured`

Reply:
> seems outside of user control entirely

There is one case where it's user-controllable (😅): they can enable or disable it via the Secret Settings menu.

We added that to make testing easier—before we had the preview experiment—but it seems like it could still be useful after that. If the experiment goes well and we roll Suggest out for everyone, I think we'll want to have `enableFxSuggest` check the current region for the default value—like `isAddressFeatureEnabled()`!—so that it's only visible and enabled for `en-US` users, but still have the secret setting as an escape hatch.

Reply:
> There is one case where it's user-controllable (😅): they can enable or disable it via the Secret Settings menu.

I am kinda of the opinion that those don't count and that we should revisit our persistence workflows there, since we don't even persist access to the menu _but_ point taken! Again since I think this work is fairly in-flight still as long as it's working as intended then it's good by me 🙂

Reply:
Gotcha, thanks! I put the (😅) because I was bemused at how complicated this is, and definitely didn't mean to make a point or anything!

"A flag that's controllable by a Nimbus feature variable now, and by a region check in the future, but can be overridden, and also affects the visibility of suggestions and Settings switches, and also also is still in-flight" is the shortest summary I can think of! 😆

Thanks for the review, and suggesting those other approaches!